wurst/objects/abilities/beastMaster/UltimateForm/Devour.wurst

```diff
@@ -0,0 +1,124 @@
+let DEVOUR_SOUND = new SoundDefinition(Sounds.devour, false, true)
+
+let HEALTH_THRESHOLD = 50
+
+let upgradeFromAnimalMap = new HashMap<int, int>()
+    ..put(UNIT_ELK, UPGD_RENDO_ELK_JUMP)
+    ..put(UNIT_HAWK, UPGD_RENDO_HAWK_EYE)
+    ..put(UNIT_JUNGLE_WOLF, UPGD_RENDO_WOLF_BITE)
+    ..put(UNIT_SNAKE, UPGD_RENDO_SNAKE_TOXIN)
+    ..put(UNIT_PANTHER, UPGD_RENDO_PANTHER_INSTINCT)
+
+let upgradeAbilityMap = new HashMap<int, int>()
+    ..put(ABILITY_RENDO_ELK_JUMP, UPGD_RENDO_ELK_JUMP)
+    ..put(ABILITY_RENDO_HAWK_EYE, UPGD_RENDO_HAWK_EYE)
+    ..put(ABILITY_RENDO_WOLF_BITE, UPGD_RENDO_WOLF_BITE)
+    ..put(ABILITY_RENDO_SNAKE_TOXIN, UPGD_RENDO_SNAKE_TOXIN)
+    ..put(ABILITY_RENDO_PANTHER_INSTINCT, UPGD_RENDO_PANTHER_INSTINCT)
+
+let COOLDOWN = 25.
+
+let TT_NORM = "Devour"
+let TT_EXT = "" +
+    "Rendo savagely devour an animal and absord its gene pool, gaining an ability based on the type of " +
+    "animal he ate. Rendo swallow up and digest every part of the animal, therefore, no hides " +
+    ", bones nor corpses are dropped upon the animal death. Rendo instantly regenerate health " +
+    "depending on which animal he devoured." +
+    makeToolTipCooldown(COOLDOWN)+
+    "\n\nFollowing animal give an ability upon devoration :\n" +
+    "Elk, Hawk, Snake, Wolf, Panther".color(GOLD_COLOR)
+
+@compiletime function createDevourSpell() returns ChannelAbilityPreset
+    return new ChannelAbilityPreset(ABILITY_RENDO_DEVOUR, 1, true)
+        ..presetTargetTypes(Targettype.UNIT)
+        ..setName(TT_NORM)
+        ..setHotkeyNormal("F")
+        ..setTooltipNormal(1, makeToolTipNorm("F", TT_NORM))
+        ..setTooltipNormalExtended(1, TT_EXT)
+        ..presetCooldown(lvl -> COOLDOWN)
+        ..presetManaCost(lvl -> 0)
+        ..presetCastRange(lvl -> 128)
+        ..presetButtonPosNormal(3, 1)
+        ..setFollowThroughTime(1, 0)
+        ..setIconNormal(Icons.bTNDevour)
+        ..setTargetsAllowed(1, commaList(
+            TargetsAllowed.enemies,
+            TargetsAllowed.nonhero,
+            TargetsAllowed.organic,
+            TargetsAllowed.nonancient
+        ))
+        ..setAnimationNames("stand")
+
+
+function onDevourCast(unit caster, unit target)
+    if target.getHP() < HEALTH_THRESHOLD
+        // Play a specific animation which... look like rendo's devouring something?
+        // Forced to pause unit because channel ability always play an animation
+        caster.pause()
+        caster.setAnimation(2)
+        caster.unpause()
+
+        let entity = UnitEntity.findForUnit(target)
+        // Check if unit eaten is an hostile entity
+        if entity instanceof Hostile
+            let hostile = entity castTo Hostile
+
+            // Disable item & corpses drop
+            hostile.setDropOndeath(false)
+
+            // Dealing 50 pure damage to kill the target
+            UnitDamageTarget(caster, target, HEALTH_THRESHOLD.toReal(), false, false, ATTACK_TYPE_CHAOS, DAMAGE_TYPE_UNIVERSAL, null)
```

Comment:
Is this guaranteed to kill the target? Can armor types be configured to resist Chaos damage, in which case we might forget about this?

Reply:
Your statement is right, I chose Chaos damage because it is designed to 100% damage on every kind of armor.

Reply:
Could just dealt something like 99999 damage, that's what they do in League of Legend when they want to make sure a target is executed.

Reply:
Yeah, that would work except that I think if an armor type reduced damage by 100% (like Mammoth did for piercing), then that still wouldn't kill the unit. I think this is fine for now but the experience library we have should just directly support killing a unit and awarding experience for a given killer without having to go through damage indirectly.

I think that the 99999 could also throw off the scoreboard, so lets keep it as is for now.